### DISCUSSION TITLE
Intelligent due-date

### DISCUSSION THREAD

**Discussion by carforge** [TechbeeAT/jtxBoard#1538] [Ideas]
I saw on apps like todoist an "intelligent" due date setting. 
For example, if someone create a new task like "Buy flowers on Friday", the task has automatically set a due date of upcoming Friday.
That would be a feature (I assume) that is pretty language dependent due to different grammar etc... 
Is something like that in scope?

Thanks & best regards

**Comment by patrickunterwegs** [comment]
Hi @carforge , I've been thinking about such a feature a few times, to do it myself would be pretty cumbersome. But what I could imagine is using AI features, meaning I could try to let the AI generate the iCalendar-structure in the background. I've tried it with ChatGPT and it actually worked well. 
However, I would need to find the time to get deeper into this topic. Also I'm a bit afraid that users would have second thoughts about it, after all the app is also meant to ensure privacy, but sending data to a random AI server, would that be acceptable...?

**Reply by carforge** [reply]
Maybe, like me 🙈, people running their ollama local AI LLMs at home. And maybe even reachable from the web.
In case someone wants to use such a feature, there could be an option to use a local running ollama instance with a small model processing these kinds of requests. But to be honest, how to train a model that generates exactly that output, I am also lacking knowledge...

But just as an idea. Utilizing local AI/LLMs (which have APIs) could be an option.

Cheers

**Reply by rogercreagh** [reply]
> Also I'm a bit afraid that users would have second thoughts about it, after all the app is also meant to ensure privacy, but sending data to a random AI server, would that be acceptable...?

Absolutely not, if you do something like this you must make it an option the user has have to turn on. Personal I see  no use for such a function - its classic tech bloat. If I type a reminder "buy flowers on friday" and I want a reminder on Friday I'll set a reminder. And if I don't want reminding, or want a reminder the day before or after I''ll do that.
